### PR TITLE
Remove RegionManager

### PR DESCRIPTION
**Pull request by paulfalgout**
Resolves https://github.com/marionettejs/backbone.marionette/issues/2743

This vastly simplifies the relationship between view and region.

It should lessen the concern about any perf-hit from all Views having regions.

### CODE REVIEW

**Comment by JSteunou** [comment]
:+1: this sounds very nice!

File: src/view.js [hunk 1]
```diff
@@ -180,64 +103,11 @@ Marionette.View = Marionette.AbstractView.extend({
...
-  _getImmediateChildren: function() {
-    return _.chain(this.regionManager.getRegions())
-      .pluck('currentView')
-      .compact()
-      .value();
+    return Marionette.AbstractView.prototype.destroy.apply(this, arguments);
   }
 });
+
+_.extend(Marionette.View.prototype, Marionette.RegionsMixin);
```
Comment (ahumphreys87) [inline src/view.js]: do we have Metal mixins yet?

**Comment by jmeas** [comment]
:heart_eyes:

File: src/view.js (hunk 1 above)
Comment (jasonLaster) [inline src/view.js]: nope, i'm going to remove metal too. It's currently flagged off, but still a headache

**Comment by jasonLaster** [comment]
I'd like to keep `actsAsCollection` it's useful and ppl might use it :)

**Comment by paulfalgout** [comment]
We should make `actsAsCollection` a separate utility maybe?  Possibly use it in babysitter as well.  Babysitter has it's own version of this, and they both get out of date and out of sync too easily.  But the utility most likely won't be used in the library at all unless babysitter is brought in.
Plus would we just attach it to the parent `Marionette`?  Seems like in ES6 world it makes more sense just to `import actAsCollection from 'backbone-actascollection';`

**Comment by paulfalgout** [comment]
@jmeas yeah I can delete the doc I suppose.. although there's lingering stuff in various places of the docs on this branch.  They're essentially unmaintainable at the moment and need a lot of love.  related https://github.com/marionettejs/backbone.marionette/issues/2477

**Comment by jmeas** [comment]
> although there's lingering stuff in various places of the docs on this branch. They're essentially unmaintainable at the moment and need a lot of love.

Gotcha. Np then; it shouldn't hope up this good work.

**Comment by paulfalgout** [comment]
@jasonLaster the previously mentioned collection function on babysitter is here: https://github.com/marionettejs/backbone.babysitter/blob/master/src/childviewcontainer.js#L135

**Comment by jasonLaster** [comment]
`import actAsCollection from 'backbone-actascollection'` sounds good to me. Lets kill it

**Comment by jasonLaster** [comment]
:+1: 

To be clear, we should merge this ASAP. PRs this good, and this big should not be allowed to go stale :bread: :egg: , else they look like :skull: which nobody wants :hear_no_evil:

**Comment by paulfalgout** [comment]
To be fair a lot of the concepts here are from various @jfairbank @jmeas discussions

**Comment by ahumphreys87** [comment]
:+1:

**Pull request merged**